newsroom/agenda/views.py

```diff
@@ -91,6 +91,13 @@ def search():
     return send_response('agenda', response)
 
 
+@blueprint.route('/agenda/count')
+@login_required
+def get_agenda_featured_item_count():
+    items = get_resource_service('agenda_featured').get_all()
```

Comment:
fetching all seems a bit expensive to realize if there are items, imo we can just try to find one and either it's there or not. also making an api call for it seems unnecessary, it could be part of the view data so you would have that info there from the start and we wouldn't need extra action/reducer for this

Reply:
I am not able to find any relevant data that could tell me if featured item exists inside `initData`. It might contain `featured` boolean variable though.
Also, there is one more thing, here I am returning the count of all the available items, irrespective of date. Do we need this toggle to be shown as per the date in the filter? Like if I select today's date, so if there are any featured items available for today, the toggle will be shown, else not?

Reply:
I mean you would need to add it there instead of doing that query.
so in this `get_view_data` function you could add:
```
    'has_agenda_featured_items': get_resource_service('agenda_featured').find_one(req=None) is not None,
```